EventFilter/EcalRawToDigi/plugins/EcalCPUDigisProducer.cc

```diff
@@ -59,8 +66,15 @@ class EcalCPUDigisProducer : public edm::stream::EDProducer<edm::ExternalWork> {
 
   // dummy integrity errors
   edm::EDPutTokenT<EcalElectronicsIdCollection> integrityTTIdErrorsToken_;
+  edm::EDPutTokenT<EcalElectronicsIdCollection> integrityZSXtalIdErrorsToken_;
```

Comment:
Have `integrityZSXtalIdErrorsToken_`, `pnDiodeDigisToken_`, `ecalTriggerPrimitivesToken_`, and `ecalPseudoStripInputsToken_` been missing before? Or why are they needed now?

Reply:
They are needed offline, while I think they are not used online.

The old configuration did not need them because it would use a completely separate workflow for the CPU and GPU case. This PR changes that to use a common workflow and let the `SwitchProducer` pick ether branch (`cpu` or `cuda`) at runtime. In turn, this requires that all branches declare (*e.g.* `produces()`) the same collections.

The alternative would be to use a set of `EDAlias`es to restrict which collections are made available via the `SwitchProducer` to only the ones produced by the GPU module. However, that would hide those collections also when running on the CPU, which is probably not what we want to do.